File: BetterCode/Palindrome.py
# Better Code

def is_palindrome(s):
    reversed_s = s[::-1]
    return s == reversed_s


input_str = "racecar"
is_pal = is_palindrome(input_str)
print(f"Is '{input_str}' a palindrome? {is_pal}")

File: BetterCode/Palindrome.java
package BetterCode;

public class Palindrome {
    public static boolean isPalindrome(String s) {
        String reversed = new StringBuilder(s).reverse().toString();
        return s.equals(reversed);
    }

    public static void main(String[] args) {
        String input = "racecar";
        boolean isPal = isPalindrome(input);
        System.out.println("Is '" + input + "' a palindrome? " + isPal);
    }
}
